core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -273,17 +275,17 @@ public Response getBucket(@HeaderParam("Authorization") String authorization,
 
       List<URIStatus> children;
       try {
-        // TODO(czhu): allow non-"/" delimiters by parsing the prefix & delimiter pair to determine
-        //             what directory to list the contents of
-        // only list the direct children if delimiter is not null
-        if (delimiterParam != null) {
+        // only list the direct children if delimiter exactly "/"
+        if (delimiterParam != null && delimiterParam.equals(AlluxioURI.SEPARATOR)) {
           if (prefixParam == null) {
             path = parsePath(path, "", delimiterParam);
           } else {
             path = parsePath(path, prefixParam, delimiterParam);
           }
           children = fs.listStatus(new AlluxioURI(path));
-        } else {
+        } else { // recursively list the path
+          // TODO(czhu): use path traversal to build list of returned children,
+          //  if a delimiter is found then you can stop traversing that sub-path
           ListStatusPOptions options = ListStatusPOptions.newBuilder().setRecursive(true).build();
           children = fs.listStatus(new AlluxioURI(path), options);
```

Comment:
Not sure why this doesn't use `parsePath` to replace the Bucket separator (":"). Will look into it.